src/Hedgehog.ts

```diff
@@ -172,8 +176,11 @@ export class Hedgehog {
         cipherText: cipherTextHex,
         lookupKey: lookupKey,
       };
+      const additionalAuthData = {
+        wallet: walletObj
+      }
 
-      await self.setAuthFn(authData);
+      await self.setAuthFn(authData, additionalAuthData);
```

Comment:
Give that `authData` has optionalParams above, couldn't we just pass it in there? Maybe instead of positional args, we should add better types to the authData object?

Reply:
I'm curious for @rickyrombo 's thoughts

Reply:
Yeah my thinking is we wouldn't need to update hedgehog at all and could just use the optional params

Reply:
Oh I get it - we don't have the wallet yet so we need to update hedgehog to pass it along. Just saw your msg, sorry!

I still think we can set a field on `authData` but you're running the risk of conflicts with `optionalParam`. I think we're ok because this is a major version change. I do think this changes the philosophy of Hedgehog significantly so when we publish it probably warrants some doc changes too (ultimately we should probably have cleaned it up to one callback at that point too...)

Reply:
Marcus is exactly right. Based on the way we've implemented the callback there's no way to get the right wallet between the time this signUp here is called and the callback is executed.
https://github.com/AudiusProject/audius-protocol/blob/main/packages/libs/src/api/Account.ts#L167-L171

I made it a second arg specifically to not expose wallets when hedgehog is used like this, where the entire obj is passed to a server. If we added the wallet object to this first arg then keys could be exposed depending on how anyones using the library. Adding it as a second param removes that possibility, adds it for users in the future, and preserves the API for backwards compatibility.
https://github.com/AudiusProject/audius-protocol/blob/main/packages/libs/src/services/identity/IdentityService.ts#L127-L133

Reply:
wondering if this becomes v4.0.0 and we just make it fully typed the correct way though (not gonna block this PR though, I think fine)
```
export type SetAuthFn = (params: {
  authData: {
    iv: string;
    cipherText: string;
    lookupKey: string;
  },
  wallet: Wallet
}

export type UpdateAuthFn = (params: {
  authData: {
    iv: string;
    cipherText: string;
    lookupKey: string;
    oldLookupKey: string
  },
  wallet: Wallet
}
```

Reply:
Do we even need to 4.0.0? we haven't published 3.0.0 yet. I think the following is fine as long as it's clear when/if we publish
```ts
export type SetAuthFn = (params: { 
  iv: string; 
  cipherText: string; 
  lookupKey: string; 
  wallet: Wallet 
}) => Promise<void>
```

Reply:
we would need to make it clear that the wallet remains client side, we could probably keep this at 3.0.0 since it's not technically released yet